integration-tests/src/test/java/tachyon/LocalTachyonClusterResource.java

```diff
@@ -0,0 +1,188 @@
+ * {@literal@}ClassRule is used), and destroyed at the end. Below is an example of declaring and
+ * using it.
+ *
+ * <pre>
+ *   public class SomethingTest {
+ *    {@literal@}Rule
+ *    public LocalTachyonClusterResource localTachyonClusterResource =
+ *      new LocalTachyonClusterResource(
+ *        WORKER_CAPACITY, QUOTA_UNIT, BLOCK_SIZE, CONF_KEY_1, CONF_VALUE_1, ...);
+ *
+ *    {@literal@}Test
+ *    public void testSomething() {
+ *      localTachyonClusterResource.get().getClient().create("/abced");
+ *      ...
+ *    }
+ *
+ *    {@literal@}Test
+ *    {@literal@}Config(tachyonConfParams = {CONF_KEY_1, CONF_VALUE_1, CONF_KEY_2,
+ *                                           CONF_VALUE_2, ...}
+ *                      startCluster = false)
+ *    public void testSomethingElse() {
+ *      localTachyonClusterResource.start();
+ *      localTachyonClusterResource.get().getClient().create("/efghi");
+ *      ...
+ *    }
+ *   }
+ * </pre>
+ */
+public class LocalTachyonClusterResource implements TestRule {
+
+  /** The capacity of the worker in bytes */
+  private final long mWorkerCapacityBytes;
+  /** The quota of space each user can request */
+  private final int mQuotaUnitBytes;
+  /** Block size for a user */
+  private final int mUserBlockSize;
+  /**
+   * If true (default), we start the cluster before running a test method. Otherwise, the method
+   * must start the cluster explicitly.
+   */
+  private final boolean mStartCluster;
+  /** Configuration parameters for the TachyonConf object used in the cluster */
+  private final String[] mConfParams;
+
+  /** The tachyon cluster being managed */
+  private LocalTachyonCluster mLocalTachyonCluster = null;
+  /** The configuration object used by the cluster */
+  private TachyonConf mTestConf = null;
+
+  /**
+   * Create a new instance.
+   *
+   * @param workerCapacityBytes the capacity of the worker in bytes
+   * @param quotaUnitBytes the quota of space each user can request
+   * @param userBlockSize the block size for a user
+   * @param startCluster whether or not to start the cluster before the test method starts
+   * @param confParams specific tachyon configuration parameters, specified as a list of strings,
+   *        treated as key-value pairs
+   */
+  public LocalTachyonClusterResource(long workerCapacityBytes, int quotaUnitBytes,
+      int userBlockSize, boolean startCluster, String... confParams) {
+    Preconditions.checkArgument(confParams.length % 2 == 0);
+    mWorkerCapacityBytes = workerCapacityBytes;
+    mQuotaUnitBytes = quotaUnitBytes;
+    mUserBlockSize = userBlockSize;
+    mStartCluster = startCluster;
+    mConfParams = confParams;
+  }
+
+  public LocalTachyonClusterResource(long workerCapacityBytes, int quotaUnitBytes,
+      int userBlockSize, boolean startCluster) {
+    this(workerCapacityBytes, quotaUnitBytes, userBlockSize, startCluster, new String[0]);
```

Comment:
Yeah, because the overloading of constructors makes it impossible to resolve otherwise.